kie-docs/jbpm-docs/src/main/docbook/en-US/Persistence/Transactions.xml

```diff
@@ -22,22 +22,34 @@
     to specify transaction boundaries, as shown below:</para>
 
     <programlisting language="java">
-// create the entity manager factory and register it in the environment
-EntityManagerFactory emf = Persistence.createEntityManagerFactory( "org.jbpm.persistence.jpa" );
-Environment env = KnowledgeBaseFactory.newEnvironment();
-env.set( EnvironmentName.ENTITY_MANAGER_FACTORY, emf );
-env.set( EnvironmentName.TRANSACTION_MANAGER, TransactionManagerServices.getTransactionManager() );
+// create the entity manager factory
+EntityManagerFactory emf = EntityManagerFactoryManager.get().getOrCreate("org.jbpm.persistence.jpa");
+TransactionManager tm = TransactionManagerServices.getTransactionManager();
+Environment env = EnvironmentFactory.newEnvironment();
+env.set(EnvironmentName.ENTITY_MANAGER_FACTORY, emf);
+env.set(EnvironmentName.TRANSACTION_MANAGER, tm);
+
+// setup the runtime environment
+RuntimeEnvironment environment = RuntimeEnvironmentBuilder.Factory.get()
+.newDefaultBuilder()
+.addAsset(ResourceFactory.newClassPathResource("MyProcessDefinition.bpmn2"), ResourceType.BPMN2)
+    .addEnvironmentEntry(EnvironmentName.TRANSACTION_MANAGER, tm)
+    .addEnvironmentEntry(EnvironmentName.PERSISTENCE_CONTEXT_MANAGER, new JpaProcessPersistenceContextManager(env))
+    .addEnvironmentEntry(EnvironmentName.TASK_PERSISTENCE_CONTEXT_MANAGER, new JPATaskPersistenceContextManager(env))
```

Comment:
no need to define manually persistence context managers, they will be setup automatically by the engine